### DISCUSSION TITLE
Docker building problems with EL9...

### DISCUSSION THREAD

**Discussion by Bunker56** [PhlexPlexico/G5API#213] [Help]
---> Running in a6c483330096
yarn install v1.22.19
warning package.json: Unable to read "./bin/" directory of module "g5api"
warning g5api@1.6.0.0: Unable to read "./bin/" directory of module "g5api"
warning g5api@1.6.0.0: Unable to read "./bin/" directory of module "g5api"
warning g5api@1.6.0.0: Unable to read "./bin/" directory of module "g5api"
[1/5] Validating package.json...
warning g5api@1.6.0.0: Unable to read "./bin/" directory of module "g5api"
[2/5] Resolving packages...
[3/5] Fetching packages...
[4/5] Linking dependencies...
warning "swagger-jsdoc > swagger-parser > @apidevtools/swagger-parser@10.0.2" has unmet peer dependency "openapi-types@>=7".
warning Workspaces can only be enabled in private projects.
[5/5] Building fresh packages...
warning Error running install script for optional dependency: "/Get5API/node_modules/cpu-features: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments: 
Directory: /Get5API/node_modules/cpu-features
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@9.0.0
gyp info using node@16.16.0 | linux | x64
gyp info find Python using Python version 3.10.5 found at \"/usr/bin/python3\"
gyp http GET https://unofficial-builds.nodejs.org/download/release/v16.16.0/node-v16.16.0-headers.tar.gz
gyp http 200 https://unofficial-builds.nodejs.org/download/release/v16.16.0/node-v16.16.0-headers.tar.gz
gyp http GET https://unofficial-builds.nodejs.org/download/release/v16.16.0/SHASUMS256.txt
gyp http 200 https://unofficial-builds.nodejs.org/download/release/v16.16.0/SHASUMS256.txt
gyp info spawn /usr/bin/python3
gyp info spawn args [
gyp info spawn args   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Get5API/node_modules/cpu-features/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/root/.cache/node-gyp/16.16.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/root/.cache/node-gyp/16.16.0',
gyp info spawn args   '-Dnode_gyp_dir=/usr/local/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/root/.cache/node-gyp/16.16.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Get5API/node_modules/cpu-features',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/Get5API/node_modules/cpu-features/build'
  ACTION Configuring dependencies /Get5API/node_modules/cpu-features/deps/cpu_features/build/Makefile
/bin/sh: cmake: not found
make: *** [config_deps.target.mk:13: /Get5API/node_modules/cpu-features/deps/cpu_features/build/Makefile] Error 127
make: Leaving directory '/Get5API/node_modules/cpu-features/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (node:events:527:28)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12)
gyp ERR! System Linux 5.14.0-70.17.1.el9_0.x86_64
gyp ERR! command \"/usr/local/bin/node\" \"/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js\" \"rebuild\"
gyp ERR! cwd /Get5API/node_modules/cpu-features
gyp ERR! node -v v16.16.0
gyp ERR! node-gyp -v v9.0.0
gyp ERR! not ok"
info This module is OPTIONAL, you can safely ignore this error
warning g5api@1.6.0.0: Unable to read "./bin/" directory of module "g5api"
warning g5api@1.6.0.0: Unable to read "./bin/" directory of module "g5api"
warning g5api@1.6.0.0: Unable to read "./bin/" directory of module "g5api"
warning g5api@1.6.0.0: Unable to read "./bin/" directory of module "g5api"
Done in 20.96s.
Removing intermediate container a6c483330096
 ---> bffa7411b79f
Step 8/9 : COPY . .
 ---> bb99eb707c3b
Step 9/9 : CMD envsubst < /Get5API/config/production.json.template > /Get5API/config/production.json  &&     yarn migrate-create-prod &&     yarn migrate-prod-upgrade &&     yarn startprod &&     yarn pm2 logs
 ---> Running in e96fb6ccb9f8
Removing intermediate container e96fb6ccb9f8
 ---> 75ba2ca7b462
Successfully built 75ba2ca7b462
Successfully tagged bunker56/g5api:latest

It`s normally? 

G5V was built without errors...

**Comment by PhlexPlexico** [comment]
Make sure you have the following installed in your Dockerfile - gettext python3 build-base - these are required for bcrypt to install on the system and allow for local logins.

**Reply by PhlexPlexico** [reply]
Also, it looks like `info This module is OPTIONAL, you can safely ignore this error` so you can ignore it, it should have built just fine :)

**Comment by Bunker56** [comment]
I made the assembly from your Dockerfile. git clone etc...

**Comment by Bunker56** [comment]
I'll check it out now.
It seems to work... I'll clarify later.

**Comment by Bunker56** [comment]
Hi! Updated the image, now this ...
and the same with your image.

G5API                   | yarn run v1.22.19
G5API                   | $ MYSQL_FLAGS="-CONNECT_WITH_DB" db-migrate --env production --config config/production.json db:create get5
G5API                   | [ERROR] SyntaxError: Unexpected token } in JSON at position 378
G5API                   |     at JSON.parse (<anonymous>)
G5API                   |     at exports.parse (/Get5API/node_modules/rc/lib/utils.js:15:17)
G5API                   |     at addConfigFile (/Get5API/node_modules/rc/index.js:30:20)
G5API                   |     at module.exports (/Get5API/node_modules/rc/index.js:46:20)
G5API                   |     at module.exports (/Get5API/node_modules/db-migrate/lib/commands/set-default-argv.js:122:47)
G5API                   |     at new dbmigrate (/Get5API/node_modules/db-migrate/api.js:75:10)
G5API                   |     at Object.module.exports.getInstance (/Get5API/node_modules/db-migrate/index.js:85:10)
G5API                   |     at /Get5API/node_modules/db-migrate/bin/db-migrate:32:25
G5API                   |     at /Get5API/node_modules/resolve/lib/async.js:142:25
G5API                   |     at maybeRealpath (/Get5API/node_modules/resolve/lib/async.js:50:9)
G5API                   | error Command failed with exit code 1.
G5API                   | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
G5API exited with code 1

**Reply by PhlexPlexico** [reply]
Please make sure you read the git commits instead of blindly upgrading :)

There is a new config option to enable/disable local logins. In your `docker-compose.yml` file please include this line right after the `UPLOADDEMOS` variable:

```
- LOCALLOGIN=true
```

Or false, if you wish to disable local register/login.

Thanks!
